### PR TITLE
more hooks

### PR DESCRIPTION
**Pull request by Emilgardis**
- add pre-push hook that tests
- fix rustfmt edition

### CODE REVIEW

**Comment by Alexhuszagh** [comment]
Might also be worth adding a shellcheck, if installed?

**Comment by Emilgardis** [comment]
oo, smart

File: xtask/src/pre-commit [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-
-set -ex
-
-flags=(--all-features --all-targets --workspace)
-cargo fmt -- --check
-if cargo +nightly >/dev/null 2>&1; then
-    cargo +nightly clippy "${flags[@]}" -- --deny warnings
-else
-    cargo clippy "${flags[@]}" -- --deny warnings
-fi
```
Comment (Emilgardis) [inline xtask/src/pre-commit:12]: why does git think this file was removed...

ill fix it

Comment (Emilgardis) [reply xtask/src/pre-commit]: ...

Comment (Alexhuszagh) [reply xtask/src/pre-commit]: No idea, it has the proper permissions at least so it seems fine.

**Review by Alexhuszagh — APPROVED** [review]

**Comment by Emilgardis** [comment]
bors r-

just a minute, think there's something wrong

File: xtask/src/pre-commit.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    cargo clippy "${flags[@]}" -- --deny warnings
+fi
+
+if shellcheck -V >/dev/null 2>&1; then
+    echo "Running shellcheck"
+    changed_shell_files=$(
+        git diff --cached --name-only --diff-filter=ACM |
+        xargs grep -lE '^#!/.*(sh|bash|ksh)'
+    )
+    if [ -n "${changed_shell_files[*]}" ]; then
```
Comment (Emilgardis) [inline xtask/src/pre-commit.sh:21]: this check is not working when there's no changed sh files

**Pull request merged**